### PR TITLE
HDDS-5777. Provide an option to dump table scan data to file.

### PR DESCRIPTION
**Pull request by bharatviswa504**
## What changes were proposed in this pull request?

Provide an option to dump entire table column family data to a file. This will be useful in HA when we want to compare DB data across SCM's/OM's for verification.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-5777

## How was this patch tested?
Added UT and also
Manually verified it on the cluster
```
ozone debug ldb --db=/var/lib/hadoop-ozone/scm/data/scm.db/ scan --column_family=containers --with-keys --out=/tmp/containers1 -l=-1
less /tmp/containers1
{
  "id": 1
} -> {
  "state": "OPEN",
  "pipelineID": {
    "id": "054f8658-81bf-4e81-a71e-4928130f5993"
  },
  "replicationConfig": {
    "replicationFactor": "THREE"
  },
  "usedBytes": 0,
  "numberOfKeys": 0,
  "lastUsed": {
    "seconds": 1632353739,
    "nanos": 455000000
  },
  "stateEnterTime": {
    "seconds": 1632337329,
    "nanos": 692000000
  },
  "owner": "om1",
  "containerID": 1,
  "deleteTransactionId": 0,
  "sequenceId": 0
}
{
  "id": 2
} -> {
  "state": "OPEN",
  "pipelineID": {
    "id": "054f8658-81bf-4e81-a71e-4928130f5993"
  },
  "replicationConfig": {
    "replicationFactor": "THREE"
  },
  "usedBytes": 0,
  "numberOfKeys": 0,
  "lastUsed": {
    "seconds": 1632353739,
    "nanos": 471000000
  },
  "stateEnterTime": {
    "seconds": 1632337343,
    "nanos": 998000000
  },
  "owner": "om1",
  "containerID": 2,
  "deleteTransactionId": 0,
  "sequenceId": 0
}
```

### CODE REVIEW

**Comment by bharatviswa504** [comment]
Fixed tests and added a test.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmLDBCli.java [hunk 1]
```diff
@@ -98,15 +103,70 @@ public void testOMDB() throws Exception {
...
     }catch (IllegalArgumentException e){
       //ignore
     }
+
+    // If set with -1, check if it dumps entire table data.
+    DBScanner.setLimit(-1);
+    Assert.assertEquals(5, getKeyNames(dbScanner).size());
+
+    // Test dump to file.
+    String outFile = "/tmp/keyTable" + LocalDate.now();
```
Comment (avijayanhwx) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmLDBCli.java:123]: Nit. Maybe better to use Java tmp dir?

**Review by avijayanhwx — COMMENTED** [review]
LGTM +1. Minor nit inline

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmLDBCli.java (hunk 1 above)
Comment (bharatviswa504) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmLDBCli.java]: Done

Comment (bharatviswa504) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmLDBCli.java]: Done

**Pull request merged**